tools/roslaunch/src/roslaunch/substitution_args.py

```diff
@@ -373,16 +377,14 @@ def resolve_args(arg_str, context=None, resolve_anon=True, filename=None):
         context = {}
     if not arg_str:
         return arg_str
-    # special handling of $(eval ...)
-    if arg_str.startswith('$(eval ') and arg_str.endswith(')'):
-        return _eval(arg_str[7:-1], context)
```

Comment:
Note that the parameters that we pass to `eval` originally were the raw string, e.g.,

```
$(eval radius * 2)
```

or

```
$eval(find('package') + '/directory')
```

They added this 'special handling' clause because you can have nested parentheses inside an `eval` tag (see the `find` example above), and `_collect_args` couldn't parse that.

To get around this, I adjusted the `_collect_args` method so it could correctly parse `eval` tags, even with nested params. Note that the splitting that we do [here](https://github.com/locusrobotics/ros_comm/pull/16/files#diff-85d266eabfa0eb4dda9a4551c1929065L403) is useless for the purposes of `eval`. All it [wants](https://github.com/locusrobotics/ros_comm/pull/16/files#diff-85d266eabfa0eb4dda9a4551c1929065R349) is the raw file string (`resolved`), the line string (`a`), and the `context`. I just had to make that method signature match the other `commands` [handlers](https://github.com/locusrobotics/ros_comm/pull/16/files#diff-85d266eabfa0eb4dda9a4551c1929065R387).